tests/unit/plugins/modules/identity/keycloak/test_keycloak_realm.py

```diff
@@ -0,0 +1,314 @@
+    if callable(object_with_future_response):
+        return object_with_future_response()
+    if isinstance(object_with_future_response, dict):
+        return get_response(
+            object_with_future_response[method], method, get_id_call_count)
+    if isinstance(object_with_future_response, list):
+        try:
+            call_number = get_id_call_count.__next__()
+        except AttributeError:
+            # manage python 2 versions.
+            call_number = get_id_call_count.next()
+        return get_response(
+            object_with_future_response[call_number], method, get_id_call_count)
+    return object_with_future_response
+
+
+def build_mocked_request(get_id_user_count, response_dict):
+    def _mocked_requests(*args, **kwargs):
+        url = args[0]
+        method = kwargs['method']
+        future_response = response_dict.get(url, None)
+        return get_response(future_response, method, get_id_user_count)
+    return _mocked_requests
+
+
+def create_wrapper(text_as_string):
+    """Allow to mock many times a call to one address.
+    Without this function, the StringIO is empty for the second call.
+    """
+    def _create_wrapper():
+        return StringIO(text_as_string)
+    return _create_wrapper
+
+
+def mock_good_connection():
+    token_response = {
+        'http://keycloak.url/auth/realms/master/protocol/openid-connect/token': create_wrapper('{"access_token": "alongtoken"}'), }
+    return patch(
+        'ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak.open_url',
+        side_effect=build_mocked_request(count(), token_response),
+        autospec=True
+    )
+
+
+class TestKeycloakRealm(ModuleTestCase):
+    def setUp(self):
+        super(TestKeycloakRealm, self).setUp()
+        self.module = keycloak_realm
+
+    def test_create_when_absent(self):
+        """Add a new realm"""
+
+        module_args = {
+            'auth_keycloak_url': 'http://keycloak.url/auth',
+            'auth_password': 'admin',
+            'auth_realm': 'master',
+            'auth_username': 'admin',
+            'auth_client_id': 'admin-cli',
+            'validate_certs': True,
+            'id': 'realm-name',
+            'realm': 'realm-name',
+            'enabled': True
+        }
+        return_value_absent = [None, {'id': 'realm-name', 'realm': 'realm-name', 'enabled': True}]
+        return_value_created = [{
+            'code': 201,
+            'id': 'realm-name',
+            'realm': 'realm-name',
+            'enabled': True
+        }]
+        changed = True
+
+        set_module_args(module_args)
+
+        # Run the module
+
+        with mock_good_connection():
+            with patch_keycloak_api(get_realm_by_id=return_value_absent, create_realm=return_value_created) \
+                    as (mock_get_realm_by_id, mock_create_realm, mock_update_realm, mock_delete_realm):
+                with self.assertRaises(AnsibleExitJson) as exec_info:
```

Comment:
since we are not doing anything with `exec_info` we might as well get rid of it - less one unused variable.

Reply:
`exec_info` is used to check if the realm has changed or not